torch/nn/modules/utils.py

```diff
@@ -57,22 +57,28 @@ def consume_prefix_in_state_dict_if_present(
         state_dict (OrderedDict): a state-dict to be loaded to the model.
         prefix (str): prefix.
     """
-    keys = sorted(state_dict.keys())
-    for key in keys:
+    keys_to_move = list(state_dict.keys())
+    for key in keys_to_move:
+        # Key starts with the prefix
         if key.startswith(prefix):
-            newkey = key[len(prefix) :]
-            state_dict[newkey] = state_dict.pop(key)
-
-    # also strip the prefix in metadata if any.
-    if "_metadata" in state_dict:
```

Comment:
Looking at the initial code, I am a little confused --

`_metadata` should be an attribute on the `state_dict()`

This can be seen here 
https://github.com/pytorch/pytorch/blob/af9b6fa04e7460b64e634b1aecd0ef32cd0d1ddc/torch/nn/modules/module.py#L1882-L1888

So if I understand correctly, this function is never correctly stripping the prefixes of `state_dict._metadata`. This looks like a bug to me

Reply:
As you pointed out, _metadata is an attribute, and yes, it looks like a bug.